libraries/Microsoft.Bot.Builder/HealthCheck.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder
+{
+    internal class HealthCheck
+    {
+        public static HealthCheckResponse CreateHealthCheckResponse(IConnectorClient connector)
+        {
+            // A derived class may override this, however, the default is that the bot is healthy given we have got to here.
+            var healthResults = new HealthResults { Success = true };
+
+            if (connector != null)
+            {
+                try
+                {
+                    // This is a mock secure SendToConversation to grab the exact HTTP headers.
+                    // If you have no appId and no secret this code will run but no produce an Authorization header.
+                    var captureHandler = new CaptureRequestHandler();
```

Comment:
This seems like it would allow a bot to be DDOSed. 

Creating the ConnectorClient is expensive, and creating a new one on each call seems like it would make a Bot fall over.

Reply:
This is not expensive - there is no IO other than the fetch token - which is unavoidable.

I did it this way so any changes (to things like user-agents) are automatically picked up. I'm guaranteeing the most consistency for what it's worth.

Actually connector clients in themselves arn't expensive - that was a misunderstanding in the original v4 code.

What is expensive is the HttpClient inside the connector client. The trick (now more firmly established) is to cache HttpClients.

Token are also expensive.

Reply:
Given that we only get the token really, it is as exposed as the rest of our stuff pretty much. Note that Azure does have DDOS protection at the network level for our app services when hosting in azure. Tokens are cached internally so even if you have a valid auth header, we'll fetch the token once. Connector clients are cached. With all of that in mind, I'm not concerned about this at this point.